### ISSUE TITLE
Exposing an existing directory

### ISSUE THREAD

**Issue by mcsilver** [BytemarkHosting/docker-webdav#44]
I want to allow access to an existing directory using webdav, but when I specify the path to that directory as the volume, webdav creates a new `data` directory in there, and that's where a webdav client can add/edit files. Is it possible to give webdav access to the directory that I specify so that clients can edit files that are already in there?

**Comment by chaomifan** [comment]
same issue

**Comment by marcovitali** [comment]
> I want to allow access to an existing directory using webdav, but when I specify the path to that directory as the volume, webdav creates a new `data` directory in there, and that's where a webdav client can add/edit files. Is it possible to give webdav access to the directory that I specify so that clients can edit files that are already in there?

You can mount this

`/path/to/folder:/var/lib/dav/data`

instead of this

`/path/to/folder:/var/lib/dav`

**Comment by mcsilver** [comment]
Thank you for the suggestion @marcovitali. For WebDAV end-users, this will suffice. I am hoping to also alleviate both backend people and backend systems from having to content with the `data` directory, as I have _existing_ directories that I want to expose via WebDAV.

**Comment by marcovitali** [comment]
> Thank you for the suggestion @marcovitali. For WebDAV end-users, this will suffice. I am hoping to also alleviate both backend people and backend systems from having to content with the `data` directory, as I have _existing_ directories that I want to expose via WebDAV.

I have an existing folder and no `data` folder. The `data` folder is inside the container, not in your file system.
I did not need to change anything in my file system, if that is your concern.

**Comment by mcsilver** [comment]
You're right, mounting as you suggested above does work! Thank you for pointing it out.

**Comment by mjnewton** [comment]
Arriving late to the party (looks like everyone's gone home!) but in case anyone else stumbles across this I thought I would add that you can mount *multiple* folders in a similar way e.g.

```text
/path/to/folder1:/var/lib/data/folder1
/path/to/folder2:/var/lib/data/folder2
```

Incidentally, I am using [another Docker image based on this one](https://hub.docker.com/r/apachewebdav/apachewebdav) which apparently has had a few updates made since the Bytemark one was seemingly left by the wayside.

**Comment by mcsilver** [comment]
@mjnewton thanks for the suggestion. I'm going to check out apachewebdav.
